src/modules/basket/components/BasketSummaryNav.tsx

```diff
@@ -0,0 +1,132 @@
+import styled from "@emotion/styled";
+import { PaperPlaneRight } from "@phosphor-icons/react";
+import { Button, ConfigProvider, Modal, Result, Typography, theme } from "antd";
+import { useState } from "react";
+type BasketSummaryNavProps = {
+  totalPrice: number;
+};
+
+const BasketSummaryNav: React.FC<BasketSummaryNavProps> = ({ totalPrice }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalStatus, setModalStatus] = useState<
+    "idle" | "pending" | "success" | "error"
+  >("idle");
+  const {
+    token: { colorPrimary, colorBgBase },
+  } = theme.useToken();
+  const openModal = () => {
+    setModalStatus("idle");
+    setIsModalOpen(true);
+  };
+  const handleOk = () => {
+    console.log("handleOk");
+    setModalStatus("pending");
+    // TODO send order to server
+    setTimeout(() => {
+      // TODO show response message before close modal
+      setModalStatus("success");
+      // setModalStatus("error");
+      setTimeout(() => {
+        setIsModalOpen(false);
+      }, 1000);
+    }, 2000);
+  };
+
+  return (
+    <>
+      <BasketSummaryNavWrapper style={{ backgroundColor: colorPrimary }}>
+        <Typography.Title
+          style={{ color: colorBgBase, margin: 0, marginLeft: "8px" }}
+          level={4}
+        >
+          Total {totalPrice} Baht
+        </Typography.Title>
+        <ConfigProvider
+          theme={{
+            components: {
+              Button: {
+                borderRadiusLG: 12,
+                paddingContentHorizontal: 20,
+              },
+            },
+          }}
+        >
+          <OrderButton
+            size="large"
+            icon={<PaperPlaneRight color={colorPrimary} size={14} />}
+            onClick={openModal}
+          >
+            <Typography.Text style={{ color: colorPrimary }}>
+              Order
+            </Typography.Text>
+          </OrderButton>
+        </ConfigProvider>
+      </BasketSummaryNavWrapper>
+      <Modal
+        centered={true}
+        open={isModalOpen}
+        maskClosable={false}
+        closable={false}
+        onOk={handleOk}
+        onCancel={() => {
+          if (modalStatus == "pending") return;
+          setIsModalOpen(false);
```

Comment:
Should we disable cancel button when modal status is pending?

Reply:
✅Done!